File: solutions/csharp/squeaky-clean/1/SqueakyClean.cs
using System.Text;

public static class Identifier
{
    public static string Clean(string identifier)
    {
        StringBuilder sb = new StringBuilder("", identifier.Length);
        for (int i = 0; i < identifier.Length; i++) {
            char c = identifier[i];
            if (Char.IsWhiteSpace(c)) sb.Append('_');
            else if (Char.IsControl(c)) sb.Append("CTRL");
            else if (c == '-') {
                sb.Append(Char.ToUpper(identifier[i + 1]));
                i++;
            }
            else if (!Char.IsLetter(c) || (c >= 'α' && c <= 'ω')) continue;
            else sb.Append(c);
        }
        return sb.ToString();
    }
}

File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        StringBuilder sb = new StringBuilder();
        Boolean flag = false;

        for (int i = 0; i < identifier.length(); i++) {
            char c = identifier.charAt(i);
            if (Character.isWhitespace(c)) sb.append('_');
            else if (c == '-') flag = true;
            else if (flag) {
                sb.append(Character.toUpperCase(c));
                flag = false;
            }
            else if (c == '0') sb.append('o');
            else if (c == '1') sb.append('l');
            else if (c == '3') sb.append('e');
            else if (c == '4') sb.append('a');
            else if (c == '7') sb.append('t');
            else if (!Character.isAlphabetic(c)) continue;
            else sb.append(c);
        }

        return sb.toString();
    }
}
